src/mxe-conf.mk

```diff
@@ -26,10 +26,10 @@ define $(PKG)_BUILD
     touch '$(CMAKE_TOOLCHAIN_DIR)/.gitkeep'
     (echo 'set(CMAKE_SYSTEM_NAME Windows)'; \
      echo 'set(MSYS 1)'; \
-     echo 'set(BUILD_SHARED_LIBS $(CMAKE_SHARED_BOOL))'; \
-     echo 'set(BUILD_STATIC_LIBS $(CMAKE_STATIC_BOOL))'; \
-     echo 'set(BUILD_SHARED $(CMAKE_SHARED_BOOL))'; \
-     echo 'set(BUILD_STATIC $(CMAKE_STATIC_BOOL))'; \
+     echo 'set(BUILD_SHARED_LIBS $(CMAKE_SHARED_BOOL) CACHE BOOL "BUILD_SHARED_LIBS" FORCE)'; \
+     echo 'set(BUILD_STATIC_LIBS $(CMAKE_STATIC_BOOL) CACHE BOOL "BUILD_STATIC_LIBS" FORCE)'; \
+     echo 'set(BUILD_SHARED $(CMAKE_SHARED_BOOL) CACHE BOOL "BUILD_SHARED" FORCE)'; \
+     echo 'set(BUILD_STATIC $(CMAKE_STATIC_BOOL) CACHE BOOL "BUILD_STATIC" FORCE)'; \
```

Comment:
> Can you elaborate on this, please?

Yes, sure.

1) There are situations when you need to prepare static library for internal usage in your project despite the fact that all third-party libraries are shared ones.
2) There are situations when you need to prepare shared library which is statically linked with third-party libraries.
3) There are (more rare) situations when you need to compile shared and static versions of your libraries simultaneously.
4) BUILD_SHARED_LIBS and BUILD_STATIC_LIBS options are commonly used for switching between different types of build. From my personal experience situations when specific options (like FREEGLUT_BUILD_STATIC_LIBS and FREEGLUT_BUILD_SHARED_LIBS) are used for this purpose are very rare.

So forcing one type of build in cmake config may be inconvenient for end users. But of course they may fix it themself...